src/ethereum/dao_fork/vm/instructions/system.py

```diff
@@ -422,6 +417,3 @@ def delegatecall(evm: Evm) -> None:
         child_evm.output[:actual_output_size],
     )
     evm.gas_left += child_evm.gas_left
```

Comment:
Another thing that I missed: In order to correctly implement gas refund from child, we should be setting child gas_left to zero:
```python
evm.gas_left += child_evm.gas_left
child_evm.gas_left = U256(0)
```
Maybe create a method called gas_refund
```python
def gas_refund(evm: Evm, child_evm: Evm) -> None:
     evm.gas_left += child_evm.gas_left
     child_evm.gas_left = U256(0)
```

Reply:
Because the child might get double counted? Does this affect logs at all?

Reply:
Because the child might get double counted? Does this affect logs at all?

Reply:
@voith ping!

Reply:
resolved offline.